node/core/pvf/src/prepare/worker.rs

```diff
@@ -27,11 +27,23 @@ use async_std::{
 	io,
 	os::unix::net::UnixStream,
 	path::{Path, PathBuf},
+	sync::Mutex,
+	task,
 };
+use cpu_time::ProcessTime;
 use parity_scale_codec::{Decode, Encode};
 use sp_core::hexdisplay::HexDisplay;
 use std::{panic, sync::Arc, time::Duration};
 
+/// A multiple of the preparation timeout (CPU time) for which we are willing to wait on the host
+/// in wall clock time. This is lenient because the wall clock can slow down up to a factor of 4
+/// (tested on 32 threads of work on an 8-core machine).
+const PREPARATION_TIMEOUT_WALL_CLOCK_FACTOR: u32 = 4;
```

Comment:
Yeah, you're right. Another way to think about it is: if we have a 5 second timeout and the job takes 2.5 s normally, under load it can take up to 5 s CPU time and 10 s wall clock time. I will keep the conservative timeout factor.